rust/osm2lanes/src/transform/tags_to_lanes/mod.rs

```diff
@@ -565,59 +568,191 @@ pub fn tags_to_lanes(
+                    "centre_turn_lanes",
+                ])),
+            });
+        }
+        // The wiki suggests that contraflow bus lanes can be specified on oneway roads
+        // let contraflow_bus = if tags.is("busway", "opposite_lane")
+        //     || tags.is("busway:left", "opposite_lane")
+        //     || tags.is("busway:right", "opposite_lane")
+        //     || tags.is("busway:both", "lane")
+        // {
+        //     1
+        // } else {
+        //     0
+        // };
+        if let Some(l) = tagged_lanes {
+            if tagged_forward.map_or(false, |f| f != l) {
+                warnings.push(RoadMsg::Ambiguous {
+                    description: None,
+                    tags: Some(tags.subset(&["oneway", "lanes", "lanes:forward"])),
+                });
+            }
+            (l, 0, 0)
         } else {
-            2
+            let assumed_forward = tagged_forward.unwrap_or(1);
+            let mut assumed_extra_bus = tagged_bus_lanes.unwrap_or(0);
+            if tags.is("busway", "lane") || tags.is("busway:both", "lane") {
+                assumed_extra_bus += 1;
+            }
+            (assumed_forward + assumed_extra_bus, 0, 0)
         }
-    } else if oneway.into() {
-        0
     } else {
-        1
-    };
-    (num_driving_fwd, num_driving_back)
+        //  busway
+        const BUSWAY: TagKey = TagKey::from("busway");
+        let mut busway_forward_lanes = 0;
+        let mut busway_backward_lanes = 0;
+        if tags.is("busway", "lane") || tags.is("busway:both", "lane") {
+            busway_forward_lanes += 1;
+            busway_backward_lanes += 1;
+        }
+        if tags.is_any(
+            BUSWAY + locale.driving_side.tag(),
+            &["lane", "opposite_lane"],
+        ) {
+            busway_forward_lanes += 1;
+        }
+        if tags.is_any(
+            BUSWAY + locale.driving_side.opposite().tag(),
+            &["lane", "opposite_lane"],
+        ) {
+            busway_backward_lanes += 1;
+        }
+
+        match (tagged_lanes, tagged_forward, tagged_backward) {
+            (_, Some(f), Some(b)) => {
+                if let Some(num_lanes) = tagged_lanes {
+                    if num_lanes != f + b + num_both_ways {
+                        warnings.push(RoadMsg::Ambiguous {
+                            description: None,
+                            tags: Some(tags.subset(&[
+                                "lanes",
+                                "lanes:forward",
+                                "lanes:both_ways",
+                                "lanes:backward",
+                            ])),
+                        });
+                    }
+                }
+                (f, num_both_ways, b)
+            }
+            (Some(l), Some(f), None) => (f, num_both_ways, l - f - num_both_ways),
+            (Some(l), None, Some(b)) => (l - b - num_both_ways, num_both_ways, b),
+            (Some(1), None, None) => (0, 1, 0),
```

Comment:
1 lane becomes a centre lane? seems fishy?

Reply:
It is a bit fishy, but I think it is because describing "how many lanes in each direction does a skinny allyway have" is fishy. I might tag them "lane_markings=no width=5", but we have to represent that in schema somehow.